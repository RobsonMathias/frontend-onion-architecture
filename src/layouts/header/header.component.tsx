import React from 'react';
import { AppBar, Container, Toolbar, Typography } from '@mui/material';

export type PageLoaderProps = {
  action?: React.ReactNode;
};

export function Header({ action }: PageLoaderProps) {
  return (
    <AppBar position="fixed" color="transparent">
      <Container maxWidth="sm">
        <Toolbar variant="regular">
          <Typography variant="h6" component="div" sx={{ flexGrow: 1 }}>
            To-Do List
          </Typography>
          {action}
        </Toolbar>
      </Container>
    </AppBar>
  );
}
