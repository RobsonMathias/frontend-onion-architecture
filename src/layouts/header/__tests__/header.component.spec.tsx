/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import React from 'react';

import { Header } from '../header.component';

describe('Header', () => {
  it('should render successfully', () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });
});
