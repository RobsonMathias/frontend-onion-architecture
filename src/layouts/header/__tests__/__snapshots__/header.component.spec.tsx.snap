// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render successfully 1`] = `
<div>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorTransparent MuiAppBar-positionFixed mui-fixed css-10srjip-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthSm css-cuefkz-MuiContainer-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
      >
        <div
          class="MuiTypography-root MuiTypography-h6 css-1juivf6-MuiTypography-root"
        >
          To-Do List
        </div>
      </div>
    </div>
  </header>
</div>
`;
